.github/workflows/twister.yaml

```diff
@@ -188,12 +188,14 @@ jobs:
             git log  --pretty=oneline | head -n 10
           fi
           echo "$HOME/.local/bin" >> $GITHUB_PATH
+          echo "$HOME/.cargo/bin" >> $GITHUB_PATH
 
           west init -l . || true
           west config manifest.group-filter -- +ci,+optional
           west config --global update.narrow true
           west update --path-cache /repo-cache/zephyrproject 2>&1 1> west.update.log || west update --path-cache /repo-cache/zephyrproject 2>&1 1> west.update.log || ( rm -rf ../modules ../bootloader ../tools && west update --path-cache /repo-cache/zephyrproject)
           west forall -c 'git reset --hard HEAD'
+          for target in `cat lib/rust/targets.txt`; do rustup target install $target; done
```

Comment:
do you expect having more rustup commands in the future? wondering if it would make sense to have these two in their own step, like having a "Rust Environment Setup", just to keep them isolated, I'd imagine this blocked would have to be copypasted to other workflows as well

Reply:
Also, isn't there a standard actions for doing this? I see few on github though they all look community maintained.

Reply:
> do you expect having more rustup commands in the future? wondering if it would make sense to have these two in their own step

agree, lets keep those in own step, later to be removed when the docker has all the toolchains.

Reply:
Yes, there will likely be another step, to pre-fetch dependencies before starting the build.

But, moving to a different step makes me leery of what I feel like I can write without any way to test it. Any idea how to actually test this without merging it?

Reply:
Oh because `pull_request_target` runs on the previous version? I would push it on https://github.com/zephyrproject-rtos/zephyr-testing/

Reply:
now that I think about it, I will be against such a change which pulls toolchains in each action invocation. I would instead focus on doing it in the docker and get this done.
Pull packages and toolchains each time we run the twister workflow both will be slower and error-prone, i.e. we will be pulling data from the network, something that can fail...

so https://github.com/zephyrproject-rtos/docker-image/pull/194 is the right approach, we just need to get CI working on that one, not sure what is going on. @stephanosio ??

Reply:
> now that I think about it, I will be against such a change which pulls toolchains in each action invocation

You could cache it though, it's no different than the python packages.

Reply:
rustup is quite reliable, python packages have broken for me in the past when python versions change sadly

Reply:
I'm ok with having the toolchains in the docker image. Keep in mind that if the `rustup` here pulls in the same set that are in the docker image, the command will do nothing, it just prints a message that it is up to date.

I was kind of intending this to be able to break the dependency between the docker image and the Zephyr tree. If in the future, we add another target that is supported, it would automatically be installed until docker could be updated. Otherwise, adding the target will depend on getting the docker image update through.

Reply:
Who can I ask about push permission on the zephyr-testing repo?

Reply:
> Who can I ask about push permission on the zephyr-testing repo?

@stephanosio @nashif? Not sure how it works, I think I have it from the release team one but maybe they can set a branch up, or maybe you can get a collab- here just for that